.github/workflows/auto-version.yml

```diff
@@ -0,0 +1,44 @@
+name: Auto Version Based on Label
+
+on:
+  pull_request:
+    branches: [main]
+    types: [closed]
+
+jobs:
+  version-bump:
+    name: 'PR Merged'
+    if: github.event.pull_request.merged == true
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Git config
+        run: |
+          git config --global user.email "${{ github.event.inputs.email }}"
+          git config --global user.name "${{ github.actor }}"
+
+      - name: Apply version bump (major)
+        if: contains(github.event.pull_request.labels.*.name, 'major')
+        run: npm version major
+
+      - name: Apply version bump (minor)
+        if: contains(github.event.pull_request.labels.*.name, 'minor')
+        run: npm version minor
+
+      - name: Apply version bump (patch)
+        if: contains(github.event.pull_request.labels.*.name, 'patch')
+        run: npm version patch
+
+      - name: Commit changes
+        run: |
+          git add -A
+          git commit -m "version bump"
```

Comment:
does this happen on the feature-branch prior to merge .. or on the main-branch post-merge?

Reply:
The intent is to run on main-branch post-merge.

Reply:
Though if it doesn't look right, please let me know! This is my first time writing one of these 😅

Reply:
Nice.
Feature request, not sure if it's possible.. https://docs.npmjs.com/cli/v9/commands/npm-publish
when we publish the new version, wondering if it's possible to label the publish with the PR's title (so devs know which version came from which change). If it's possible, no worries if it's not.
Clearly that would be a separate effort.

Reply:
i believe if we make the bump part of the feature branch - we may run into race conditions if there are parallel PRs being merged.
main branch would be more fool proof since it will bump on top of latest changes.

Reply:
basically two PRs merging at the same time may get the same bumped value...
i think main is better for sure.. and bots bump should be ok too maybe.. if we want to pick up a new version based off an upgrade,

Reply:
ah, does this mean I should change the checkout to:
```yaml
- uses: actions/checkout@v3
  with:
    ref: main
```
because according to the documentation it will default to checking out the branch of the event that triggered the action, which might be the feature branch if this comes from a PR?

Reply:
ah got it.. if this happens post-merge then yes maybe we should use `main` to avoid competing PRs

Reply:
we would have to test it.. with echo/print statements maybe..
there's this https://terraformationteam.slack.com/archives/C01C69KA23Y/p1671532900156529

i used a personal repo to test some of these semantics last time (without actual publishes)

Reply:
I was probably wrong in assuming `/main` won't have race conditions. We could potentially have race conditions on main as well. We would need to, at some point, merge the bump and publish scripts into one,, if publish fails, bump again and retry. For now this is a great.

Reply:
Right now we could potentially have two merges bumping to the same version (if they are merged at the same time).
I think only one of them will get published by the publish script (since it won't see any changes in version on npm, when the second bump is committed).

Reply:
Ah, I see what you mean. Yeah that would be a good improvement.